src/cascadia/TerminalApp/TerminalWindow.cpp

```diff
@@ -141,6 +136,10 @@ namespace winrt::TerminalApp::implementation
     // - Implements the IInitializeWithWindow interface from shobjidl_core.
     HRESULT TerminalWindow::Initialize(HWND hwnd)
     {
+        _root = winrt::make_self<TerminalPage>();
+        _root->WindowProperties(*this);

```

Comment:
Well this is definitely the circular ref that `TerminalPage` has up to the `TerminalWindow`. There's still a leak though even if we remove this and get a `TerminalWindow` dtor. Where....?

Reply:
* The TerminalPage does get dtor'd in `main`
* Commenting all this `WindowProperties` grabage does result in `TerminalWindow` dtoring, but not TerminalPage
* Closing a _second-to-last_ tab does result in the tab/TermControl/ControlCore/AtlasEngine dtoring. **But not the very last tab.** 
* [ ] `~TerminalPage` _doesn't fire at all_ in `foreword`, #14825. But `~ControlCore` does (for the last tab)
* ...

Reply:
guess what's SHOCKINGLY LOAD BEARING

Reply:
https://github.com/microsoft/terminal/blob/c08742ff18bbab51fb34bb0954718efa3567cbd0/src/cascadia/TerminalApp/App.cpp#L64-L70

A XAML island thread needs that rundown so it can finish propagating all it's events, including the Closing events we use to tear down the UI tree.